### PR TITLE
fix(rest-connector): allow sending formdata on put request

### PR DESCRIPTION
**Pull request by kevinkowa**
this fixes an issue with NLU when trying to send a file through a put request. Otherwise, it would
change the put request to a post automatically.

### CODE REVIEW

File: Connection/RESTConnector.cs [hunk 1]
```diff
@@ -488,6 +488,10 @@ private IEnumerator ProcessRequestQueue()
                             Log.Error("RESTConnector.ProcessRequestQueue()", "Exception when initializing WWWForm: {0}", e.ToString());
                         }
                         unityWebRequest = UnityWebRequest.Post(url, form);
+                        if (req.HttpMethod == UnityWebRequest.kHttpVerbPUT) 
+                        {
+                            unityWebRequest.method = "PUT";
+                        }
```
Comment (mediumTaj) [inline Connection/RESTConnector.cs:494]: This seems a bit inconsistent btw post and put

does this work?

```cs
if (req.HttpMethod == UnityWebRequest.kHttpVerbPOST)
{
    unityWebRequest = UnityWebRequest.Post(url, form);
}
if (req.HttpMethod == UnityWebRequest.kHttpVerbPUT)
{
    unityWebRequest = UnityWebRequest.Put(url, form);
}

```

Comment (kevinkowa) [reply Connection/RESTConnector.cs:494]: @mediumTaj Put doesn't allow formdata in the second parameter

**Review by mediumTaj — APPROVED** [review]

**Review by jeff-arn — APPROVED** [review]

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**